chalice/app.py

```diff
@@ -932,3 +954,36 @@ def __init__(self, func, name, handler_string):
 
     def __call__(self, event, context):
         return self.func(event, context)
+
+
+class S3EventConfig(object):
+    def __init__(self, name, bucket, events, prefix, suffix, handler_string):
+        self.name = name
+        self.bucket = bucket
+        self.events = events
```

Comment:
Playing around with the S3 API, it looks like it is possible to have two different lambda configurations to the same lambda function. Specifically, using the CLI some thing like this:
```
$ aws s3api get-bucket-notification-configuration --bucket chalice-s3-notify
{
    "LambdaFunctionConfigurations": [
        {
            "LambdaFunctionArn": "arn:aws:lambda:us-west-2:934212987125:function:s3-notify-dev-index", 
            "Id": "OWQxODY5Y2YtZjA3MC00M2IyLTgwZDItYTg4ODA3MmZlYjdl", 
            "Events": [
                "s3:ObjectCreated:*"
            ]
        }, 
        {
            "Filter": {
                "Key": {
                    "FilterRules": [
                        {
                            "Name": "Prefix", 
                            "Value": "some-prefix"
                        }
                    ]
                }
            }, 
            "LambdaFunctionArn": "arn:aws:lambda:us-west-2:934212987125:function:s3-notify-dev-index", 
            "Id": "YzQwNzc4OTktYzUzNi00ZjAyLWFhYTItOTBlNGNlNWYyN2E1", 
            "Events": [
                "s3:ObjectRemoved:*"
            ]
        }
    ]
}
```
Currently, we cannot support this configuration through the `on_s3_event()` because prefix and suffix rules are applied to a single list of events. To get this to work, it seems like we would need to expose a `configurations` parameter that allows to specify events and suffixes/prefixes per element in the `configurations` list. Does it make sense to support this? The one downside I see is that it makes it a more complicated interface.

Reply:
Yeah seems reasonable.